File: sdks/sdkwork-craw-chat-sdk/sdkwork-craw-chat-sdk-flutter/composed/lib/src/presence_module.dart
import 'package:backend_sdk/backend_sdk.dart';

import 'context.dart';

class CrawChatPresenceModule {
  final CrawChatSdkContext context;

  CrawChatPresenceModule(this.context);

  Future<PresenceSnapshotView?> heartbeat(PresenceDeviceRequest body) {
    return context.backendClient.presence.heartbeat(body);
  }

  Future<PresenceSnapshotView?> current() {
    return context.backendClient.presence.getPresenceMe();
  }
}

File: sdks/sdkwork-craw-chat-sdk/sdkwork-craw-chat-sdk-typescript/composed/src/presence-module.ts
import type {
  PresenceDeviceRequest,
  PresenceSnapshotView,
} from './types.js';
import type { CrawChatSdkContext } from './sdk-context.js';

export class CrawChatPresenceModule {
  constructor(private readonly context: CrawChatSdkContext) {}

  heartbeat(body: PresenceDeviceRequest): Promise<PresenceSnapshotView> {
    return this.context.backendClient.presence.heartbeat(body);
  }

  current(): Promise<PresenceSnapshotView> {
    return this.context.backendClient.presence.getPresenceMe();
  }
}
